compiler/src/dotty/tools/dotc/transform/init/Trace.scala

```diff
@@ -0,0 +1,76 @@
+package dotty.tools.dotc
+package transform
+package init
+
+import core.*
+import Contexts.*
+import ast.tpd.*
+import util.SourcePosition
+
+import Decorators._, printing.SyntaxHighlighting
+
+import scala.collection.mutable
+
+object Trace:
```

Comment:
A high-level comment would be useful here about what a `Trace` "is", and specifically explaining how a `Trace` relates to an `Error`. Specifically, say whether a trace is specifically a (call) stack trace, or just a list of positions that may or may not correspond to method calls.